### PR TITLE
 Remove unused dependencies

### PR DESCRIPTION
**Pull request by sinkuu**

### CODE REVIEW

File: crates/ra_syntax/Cargo.toml [hunk 1]
```diff
@@ -15,10 +15,6 @@ rustc-hash = "1.0.1"
 arrayvec = "0.4.10"
 once_cell = "1.2.0"
 
-# ideally, `serde` should be enabled by `ra_lsp_server`, but we enable it here
-# to reduce number of compilations
-smol_str = { version = "0.1.12", features = ["serde"] }
```
[the remark this answers is not in the document]
Comment (sinkuu) [reply crates/ra_syntax/Cargo.toml]: Right, I missed that this crate transitively depends on `smol_str`.

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by lnicola** [comment]
Maybe it's a stupid question, but what's that thing about `smol_str` and `serde`? Does `smol_str` get compiled twice, once with the feature and once without?

**Comment by matklad** [comment]
If you `cd crate/ra_syntax && cargo build`, `ra_syntax` will be build with smol_str without serde. If you `cd crate/ra_lsp_server && cargo build`, `ra_syntax` will be build again, with smol_str with serde

**Comment by lnicola** [comment]
But why not `cargo build -p ra_syntax`?

**Comment by matklad** [comment]
I think that would cause the same effect

**Comment by lnicola** [comment]
Ah, you're right :confused:. I thought it would use the same features as requested by the entire workspace, not only by that crate in isolation.

**Pull request merged**